src/modules/units/hooks/useUnitCeActions.tsx

```diff
@@ -51,32 +50,28 @@ export const useUnitCeActions = ({
         }),
       });
 
-      // Opportunity creation is only available if the unit has an associated CE Workflow Template
-      const hasWorkflowTemplate = unit.workflowTemplateName;
+      if (project.access.canManageUnits) {
+        if (unit.canBeMarkedAvailable) {
+          // TODO(#7537) - canBeMarkedAvailable doesn't guarantee that there are no current occupants.
+          // Implement a confirmation modal enabling the user to specify the "available on date".
+          actions.push({
+            title: 'Mark as Available for Referrals',
+            key: 'markAvailable',
+            ariaLabel: `Mark Unit ${unit.id} as Available for Referrals`,
+            onClick: () => {
+              markUnitsAvailable({ variables: { unitIds: [unit.id] } });
+            },
+          });
+        }
 
-      if (hasWorkflowTemplate && project.access.canManageUnits) {
-        if (unit.latestOpportunity && unit.latestOpportunity.active) {
-          // Show this option if the opportunity is active, but disable it if it's locked
+        if (unit.canBeMarkedUnavailable) {
           actions.push({
             title: 'Mark as Unavailable for Referrals',
             key: 'markUnavailable',
             ariaLabel: `Mark Unit ${unit.id} as Unavailable for Referrals`,
             onClick: () => {
               markUnitsUnavailable({ variables: { unitIds: [unit.id] } });
             },
-            disabled:
-              unit.latestOpportunity.status === CeOpportunityStatus.Locked,
-            disabledReason:
```

Comment:
I want to call out that, thanks to using the new helpers from the backend, we _are_ losing some UI explanatory text: some actions that would have formerly been disabled with a `disabledReason` will instead no longer appear.

My feeling is that the previous logic was already hard to follow (and therefore hard to maintain), and it's a worthwhile tradeoff to keep it in the backend. Hopefully the columns "Occupancy" and "Referral Status" help with the user's understanding of why the action may be missing. But let me know what you think!

Reply:
agreed!